app/model/service.py

```diff
@@ -131,14 +131,28 @@ def TryCheckStatus(self):
 		status = 0
 		resultMessage = ''
 		try:
-			if self.data[Service.STATUS_CHECK_STRING]['type'] == 'HTTP':
+			status_check_type = self.data[Service.STATUS_CHECK_STRING].get('type', None)
+			if status_check_type == 'HTTP':
 				r = self.Request()
 				if r != None:
 					status = r.status_code
 					if status != 200:
 						resultMessage = r.text
+			elif status_check_type == 'Heartbeat':
+				required_count = self.data[Service.STATUS_CHECK_STRING]['count'] #how many events...
+				required_type = self.data[Service.STATUS_CHECK_STRING].get('event_type', 'Heartbeat') # of this type
+				tag = self.data[Service.STATUS_CHECK_STRING].get('tag', None) # and possibly with the given tag in the message
+				period = int(self.data[Service.STATUS_CHECK_STRING]['minutes']) #in how many minutes do we expect
+				query = {'type': required_type, 
+						'time': {'$gt': datetime.datetime.now() - datetime.timedelta(minutes=period)}}
+				if tag:
+					query['tags'] = tag
+				count = Event.Count(self.id, query)
```

Comment:
Might want to be very careful with this, this could be super slow on the prod db.
It's all fine for a start, but I think it would make more sense to check those heartbeats when actually receiving the events (maybe in the event class on insert?)